cmd/faFindFast/faFindFast.go

```diff
@@ -6,29 +6,69 @@ package main
 import (
 	"flag"
 	"fmt"
+	"github.com/vertgenlab/gonomics/dna"
 	"log"
 	"math"
+	"os"
+	"runtime/pprof"
 
-	"github.com/vertgenlab/gonomics/exception"
 	"github.com/vertgenlab/gonomics/fasta"
-	"github.com/vertgenlab/gonomics/fileio"
 )
 
-func faFindFast(inFile string, outFile string, windowSize int, chromName string, removeN bool, longOutput bool, divergenceRate float64) {
-	records := fasta.Read(inFile)
+type Settings struct {
+	InFile          string
+	OutFile         string
+	FirstQueryName  string
+	SecondQueryName string
+	WindowSize      int
+	RefChromName    string
+	RemoveN         bool
+	LongOutput      bool
+	DivergenceRate  float64
+	OutputAlnPos    bool
+}
+
+func faFindFast(s Settings) {
+	var reference, firstQuery, secondQuery []dna.Base
+	var found bool
 
-	if len(records) != 2 {
-		log.Fatalf("Wrong number of sequences, expecting two, found %d.\n", len(records))
+	records := fasta.Read(s.InFile)    // fasta.Read already makes sure that sequence names are unique
+	recordsMap := fasta.ToMap(records) // generate map[string][]dna.Base. No order but can do key-value search
+
+	if len(records) < 2 {
+		log.Fatalf("Error: There must be at least 2 fasta records in the input file.\n")
+	}
+
+	//if reference and query names were not specified in the command, then use 1st and 2nd fasta records' names, respectively
+	//allows backward compatibility with previously-written code
+	if s.FirstQueryName == "" {
```

Comment:
#### For setting the firstQuery and secondQuery, I'd get rid of the redundancy code. We can separate this logic into a helper function like this:

```golang
func getSequenceFromName(recordsMap map[string][]dna.Base, name string, defaultSeq []dna.Base) []dna.Base {
	if name == "" {
		return defaultSeq
	}
	seq, exists := recordsMap[name]
	if !exists {
		log.Fatalf("Error: %s name is not found in the input file.\n", name)
	}
	return seq
}
```
#### Once you have helper function you can just call `getSequenceName()` and set `firstQuery` and `secondQuery`:

```suggestion
	firstQuery = getSequenceFromName(recordsMap, s.FirstQueryName, records[0].Seq)
	secondQuery = getSequenceFromName(recordsMap, s.SecondQueryName, records[1].Seq)
```